src/transformers/models/dac/modeling_dac.py

```diff
@@ -0,0 +1,653 @@
+# coding=utf-8
+# Copyright 2024 Descript and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transformers DAC model."""
+
+import math
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.utils import weight_norm
+
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    ModelOutput,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+from .configuration_dac import DacConfig
+
+
+# General docstring
+_CONFIG_FOR_DOC = "DacConfig"
+
+
+@dataclass
+class DacOutput(ModelOutput):
+    """
+    Args:
+        loss (`torch.Tensor`):
+            Loss from the encoder model, comprising the weighted combination of the commitment and codebook losses.
+        audio_values (`torch.Tensor` of shape `(batch_size, 1, input_length)`):
+            Decoded audio data.
+        quantized_representation (`torch.Tensor` of shape `(batch_size, dimension, time_steps)`):
+            Quantized continuous representation of input.
+        codebook_indices (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`):
+            Codebook indices for each codebook (quantized discrete representation of input).
+        projected_latents (`torch.Tensor` of shape `(batch_size, num_codebooks * dimension, time_steps)`):
+            Projected latents (continuous representation of input before quantization).
+    """
+
+    loss: torch.FloatTensor = None
+    audio_values: torch.FloatTensor = None
+    quantized_representation: torch.FloatTensor = None
+    codebook_indices: torch.FloatTensor = None
```

Comment:
Let's try to keep the same naming as Encodec, at least for the codebook indices ! 

Let's do [`audio_codes`](https://github.com/huggingface/transformers/blob/e65502951593a76844e872fee9c56b805598538a/src/transformers/models/encodec/modeling_encodec.py#L67) instead of `codebook_indices` ? WDYT ?

Reply:
LGTM!